### PR TITLE
Harden Codex cost scanner fork and refresh handling

### PR DESCRIPTION
**Pull request by ratulsarna**
## Summary

Tighten the Codex cost scanner around fork inheritance, cold-cache discovery, and sessions-root changes.

This keeps the fork replay fix from `#680`, but also addresses follow-on edge cases in incremental parsing, cache reuse, and long-lived session discovery.

Supersedes #680.

Thanks to @xx205 for the original fix and test coverage that made this follow-up much easier.

## Validation

- `pnpm check`
- `swift test --filter CostUsageScannerBreakdownTests --filter CostUsageScannerTests`
- `./Scripts/compile_and_run.sh`

### CODE REVIEW

File: Sources/CodexBarCore/Vendored/CostUsage/CostUsageScanner.swift [hunk 1]
```diff
@@ -503,34 +984,75 @@ enum CostUsageScanner {
...
+                for fileURL in rootFiles.sorted(by: { $0.path < $1.path }) where !seenPaths.contains(fileURL.path) {
+                    seenPaths.insert(fileURL.path)
+                    files.append(fileURL)
+                }
+
+                if !includeRecursive, shouldRunColdCacheLookback, let coldCacheLookbackStart {
+                    let recentlyModifiedFiles = Self.listCodexRecentlyModifiedFiles(
+                        root: root,
+                        modifiedSince: coldCacheLookbackStart)
+                    for fileURL in recentlyModifiedFiles.sorted(by: { $0.path < $1.path })
```
[the remark this answers is not in the document]
Comment (ratulsarna) [reply Sources/CodexBarCore/Vendored/CostUsage/CostUsageScanner.swift:1021]: Leaving this as-is for now. It’s a real but low-impact cache-hygiene/perf edge case: empty-day files from the cold lookback don’t affect reported usage and won’t be pulled back in by the overlapping-day cache path. I’d rather keep this PR focused on the correctness fixes and revisit this separately if it shows up in practice.
